PairedEndSingleSampleWf-fc-b37.wdl

```diff
@@ -0,0 +1,1753 @@
+  runtime {
+    disks: "local-disk " + sub(disk_size, "\\..*", "") + " HDD"
+    memory: "3 GB"
+    preemptible: preemptible_tries
+    docker: docker
+  }
+  output {
+    File metrics = "${metrics_filename}"
+  }
+}
+
+# Get version of BWA
+task GetBwaVersion {
+  String docker
+  String gotc_path
+
+  command {
+    # not setting set -o pipefail here because /bwa has a rc=1 and we dont want to allow rc=1 to succeed because
+    # the sed may also fail with that error and that is something we actually want to fail on.
+    ${gotc_path}bwa 2>&1 | \
+    grep -e '^Version' | \
+    sed 's/Version: //'
+  }
+  runtime {
+    memory: "1 GB"
+    docker: docker
+  }
+  output {
+    String version = read_string(stdout())
+  }
+}
+
+# Read unmapped BAM, convert on-the-fly to FASTQ and stream to BWA MEM for alignment, then stream to MergeBamAlignment
+task SamToFastqAndBwaMemAndMba {
+  File input_bam
+  String bwa_commandline
+  String bwa_version
+  String output_bam_basename
+  File ref_fasta
+  File ref_fasta_index
+  File ref_dict
+
+  # This is the .alt file from bwa-kit (https://github.com/lh3/bwa/tree/master/bwakit),
+  # listing the reference contigs that are "alternative".
+  File? ref_alt
+
+  File ref_amb
+  File ref_ann
+  File ref_bwt
+  File ref_pac
+  File ref_sa
+  Float disk_size
+  Int compression_level
+  Int preemptible_tries
+
+  String docker
+  String gotc_path
+
+  command <<<
+    set -o pipefail
+    set -e
+
+    # set the bash variable needed for the command-line
+    bash_ref_fasta=${ref_fasta}
+    
+      java -Xms5000m -jar ${gotc_path}picard.jar \
+        SamToFastq \
+        INPUT=${input_bam} \
+        FASTQ=/dev/stdout \
+        INTERLEAVE=true \
+        NON_PF=true | \
+      ${gotc_path}${bwa_commandline} /dev/stdin - 2> >(tee ${output_bam_basename}.bwa.stderr.log >&2) | \
+      java -Dsamjdk.compression_level=${compression_level} -Xms3000m -jar ${gotc_path}picard.jar \
+        MergeBamAlignment \
+        VALIDATION_STRINGENCY=SILENT \
+        EXPECTED_ORIENTATIONS=FR \
+        ATTRIBUTES_TO_RETAIN=X0 \
+        ATTRIBUTES_TO_REMOVE=NM \
+        ATTRIBUTES_TO_REMOVE=MD \
+        ALIGNED_BAM=/dev/stdin \
+        UNMAPPED_BAM=${input_bam} \
+        OUTPUT=${output_bam_basename}.bam \
+        REFERENCE_SEQUENCE=${ref_fasta} \
+        PAIRED_RUN=true \
+        SORT_ORDER="unsorted" \
+        IS_BISULFITE_SEQUENCE=false \
+        ALIGNED_READS_ONLY=false \
+        CLIP_ADAPTERS=false \
+        MAX_RECORDS_IN_RAM=2000000 \
+        ADD_MATE_CIGAR=true \
+        MAX_INSERTIONS_OR_DELETIONS=-1 \
+        PRIMARY_ALIGNMENT_STRATEGY=MostDistant \
+        PROGRAM_RECORD_ID="bwamem" \
+        PROGRAM_GROUP_VERSION="${bwa_version}" \
+        PROGRAM_GROUP_COMMAND_LINE="${bwa_commandline}" \
+        PROGRAM_GROUP_NAME="bwamem" \
+        UNMAPPED_READ_STRATEGY=COPY_TO_TAG \
+        ALIGNER_PROPER_PAIR_FLAGS=true \
+        UNMAP_CONTAMINANT_READS=true \
+        ADD_PG_TAG_TO_READS=false
+  >>>
```

Comment:
The diff says you removed these lines  from the production WDL:
```
grep -m1 "read .* ALT contigs" ${output_bam_basename}.bwa.stderr.log | \ 
grep -v "read 0 ALT contigs"
```
Any reason for that?

Reply:
b37 doesn't have an .alt file in its references so its removed for this file.